### PR TITLE
Toggles harddel searching for use on live [DNM]

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

I'm interested in fixing some of the more obscure hard deletes, namely new_player, monkeys, and some other smaller stuff.

I can't replicate them on a local though, so why not test on live to find out what's holding all the refs?

It's not suited for long term use, since the code it uses is VERY slow. So if a huge source of harddeletes is created this would cause OOMS as we wouldn't be able to churn through them fast enough.
However since things are quite stable at the moment, I'd like to do some splunking.

Enables reference tracking on failure to del, replaces testing() with log_world so the data gathered from this test is usable, makes the reference tracker check tick, which ought to make this suitable for live testing.

(If you merge this you deserve everything that happens to you)

## Why It's Good For The Game

It isn't quite yet

### CODE REVIEW

**Comment by Colovorat** [comment]
Speed merge it

**Comment by redguy999** [comment]
This PR was testmerged on Manuel, it gave everyone the debug panel, and also caused the server to crash, it seems.

**Comment by LemonInTheDark** [comment]
needs a bit of work yeah

**Comment by LemonInTheDark** [comment]
This contains <https://github.com/tgstation/tgstation/pull/59299>, just for testing reasons. I'll pull it out when I pr the other fixes, it deserves its own pr

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -1581,6 +1581,12 @@
 	color = "#FFFFFF00" // white + no alpha
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
+/datum/reagent/colorful_reagent/powder/invisible/Destroy()
+	//WEEE WOOOOOO REMOVE THIS BEFORE A MERGEEEEEEEEEEEEEEEEEEEEEE
+	stack_trace("Crayon deleted, what caused it? There's a harddel in here somewhere")
```
Comment (AnturK) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1586]: I'm just going to remind you now.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/machinery/camera/tracking.dm [hunk 2]
```diff
@@ -49,9 +49,9 @@
 		track.namecounts[name] = 1
 
 		if(ishuman(L))
-			track.humans[name] = L
+			track.humans[name] = WEAKREF(L)
 		else
-			track.others[name] = L
+			track.others[name] = WEAKREF(L)
```
Comment (AnturK) [inline code/game/machinery/camera/tracking.dm:54]: This causes runtimes in ai Topics related to tracking where it expects mobs instead of weakrefs.

Comment (LemonInTheDark) [reply code/game/machinery/camera/tracking.dm:54]: gottcha

Comment (LemonInTheDark) [reply code/game/machinery/camera/tracking.dm:54]: Should be fixed now

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:1586]: :)

**Comment by LemonInTheDark** [comment]
I'd call this a success, I may return to this idea in the future, but I think this is a pretty damn good haul for now. See #59371 for the version intended to be merged

**Pull request closed without merging**